packages/super-components/generator/component/index.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable import/extensions */
+import { dirname, join } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { readdirSync, writeFileSync } from 'node:fs';
+
+const currentDirectory = dirname(fileURLToPath(import.meta.url));
+
+const currentComponentList = readdirSync(
+  join(currentDirectory, '../../components/'),
+  {
+    encoding: 'utf-8',
+    withFileTypes: true,
+  },
+)
+  .filter((folder) => folder.isDirectory())
+  .map((folder) => folder.name);
+
+const getSuperComponentSetupContent = (
+  componentTag,
+) => `/* DO NOT MODIFY: this file is generated by the super-component generator */
+${JSON.stringify([
+  ...currentComponentList,
+  componentTag,
+])}.forEach((component) =>
+  import(
+    \`../../components/\${component}/loader/index.js\`
+  ).then(({ defineCustomElements }) => defineCustomElements()),
+);
+`;
+
+export default (plop) => {
+  plop.setGenerator('component', {
+    description: 'Create a new super-component',
+    prompts: [
+      {
+        type: 'input',
+        name: 'name',
+        message: 'What is the name of the new component ?',
+        default: 'Manager Hello',
+        validate: (name) => name.length > 1,
+      },
+      {
+        type: 'input',
+        name: 'componentTag',
+        message: 'What is the tag of the new component ?',
+        default: 'msc-hello',
+        validate: (componentTag) =>
+          componentTag.length > 1 &&
+          !currentComponentList.includes(componentTag),
```

Comment:
**question:** Are we restricting all the super-component tag to start with `msc-`. 
If yes, we can add this check to validate function.
If not, we should consider restricting.